resources/analysisTools/tools/annotate_vcf.pl

```diff
@@ -296,7 +296,11 @@ BEGIN
         @b_lines = ();
         $next_b_line = {};
         close $b_fh if (ref($b_fh));
-        open $b_fh, TABIX_BIN . " $opts{bfile} ${b_chr_prefix}${chr}${b_chr_suffix} |" or die "opening b file $opts{bfile} with tabix failed";
+        if($chr=~/^HLA/){
+            open $b_fh, TABIX_BIN . " $opts{bfile} ${chr}: |" or die "opening b file $opts{bfile} with tabix failed";
```

Comment:
Why is there a colon `:` following the chromosome name? Just wondering, I am not familiar with tabix. A colon is possible with an interval afterwards, but does it also work without? What happens if there is no colon? I guess the same (and not s.th. weird like a prefix-match).

> BTW: I think in the new workflow versions we should use metadata tables for the chromosomes. Something like with the fields `name`, `target_name`, `isHLA`, `includeQC`, etc. Maybe also an annotation in a YAML/JSON config file instad of a table. This will enormously simplify the code and make it less dependent on the particular choice of chromosome names.

Reply:
We need the `:` for the HLA contigs because the contig names in the reference files are like this, `HLA-A*01:01:01:01`. Normally to extract all the variants from the `chr1`, we do `tabix $file chr1`, but this doesn't seem to work for HLA contigs, we need `tabix $file HLA-A*01:01:01:01:`

Reply:
I would lie if I said I understand what tabix is doing there. I can only guess tabix's region parser takes the last `:` and considers everything before the contig name, and that it can then also handle empty start and end positions (because they are missing in your call).

If that is true, then the same should work for `chr1`. In that case it would be better (clearer) not to conflate the HLA-colon-logic with the prefix/suffix logic. What about prefixed/suffixed contig names? Are all contigs usually prefixed/suffixed (including the _alt), except for the HLA contigs?

Maybe s.th. like the following would be possible:

```bash
if ($chr=~/^HLA/) {
   chr_mod=$chr;
} else {
   chr_mod="${b_chr_prefix}${chr}${b_chr_suffix}";
}
open $b_fh, TABIX_BIN . " $opts{bfile} ${chr_mod}: |" 
  or die "opening b file $opts{bfile} with tabix failed";
```

Reply:
Colon at the end would work for other chr as well, and it is also neat so I will add these changes. Since we have the mix of 'chr' prefixes in the hg38 reference, we are encountering this issue.